### ISSUE TITLE
ERROR: Could not find a valid gem 'deep_merge'

### ISSUE THREAD

**Issue by CraigMason** [puphpet/puphpet#2069]
Downloaded a new config tonight and added some minor additions. Droplet gets created, but cannot provision:

```
==> default: ERROR:  Could not find a valid gem 'deep_merge' (>= 0), here is why:
==> default:           Unable to download data from https://rubygems.org/ - bad response Not Found 404 (https://rubygems.global.ssl.fastly.net/latest_specs.4.8.gz)
==> default: ERROR:  Possible alternatives: deep_merge
==> default: ERROR:  Could not find a valid gem 'activesupport' (>= 0), here is why:
==> default:           Unable to download data from https://rubygems.org/ - bad response Not Found 404 (https://rubygems.global.ssl.fastly.net/latest_specs.4.8.gz)
==> default: ERROR:  Possible alternatives: activesupport
==> default: ERROR:  Could not find a valid gem 'vine' (>= 0), here is why:
==> default:           Unable to download data from https://rubygems.org/ - bad response Not Found 404 (https://rubygems.global.ssl.fastly.net/latest_specs.4.8.gz)
```

Vagrant 1.8.1, vagrant-digitalocean (0.7.8)

Config:

```yaml
vagrantfile:
    target: digitalocean
    vm:
        box: digital_ocean
        hostname: mysite-test.somedomain.com
        network:
            private_network: 192.168.56.101
            forwarded_port: {  }
        provider:
            digital_ocean:
                image: ubuntu-14-04-x64
                region: sfo1
                size: 512mb
                token: somelongtoken
                ssh_key_name: craig_rmbp
        provision:
            puppet:
                manifests_path: puphpet/puppet/manifests
                module_path: puphpet/puppet/modules
                options:
                    - '--verbose'
                    - '--hiera_config /vagrant/puphpet/puppet/hiera.yaml'
        synced_folder: {  }
    ssh:
        host: null
        port: null
        private_key_path: ~/.ssh/id_rsa
        username: myuser
        guest_port: null
        keep_alive: true
        forward_agent: false
        forward_x11: false
        shell: 'bash -l'
        insert_key: false
    vagrant:
        host: detect
server:
    install: '1'
    packages:
        - vim
        - htop
        - tmux
        - nfs-kernel-server
        - acl
        - nodejs-legacy
users_groups:
    install: '1'
    groups: {  }
    users:
        - '{mysite}:{mysite}'
locale:
    install: '1'
    settings:
        default_locale: en_GB.UTF-8
        locales:
            - en_GB.UTF-8
            - en_US.UTF-8
firewall:
    install: '0'
    rules: {}
cron:
    install: '1'
    jobs: {  }
nginx:
    install: '1'
    settings:
        default_vhost: 1
        proxy_buffer_size: 128k
        proxy_buffers: '4 256k'
        proxy_connect_timeout: 600s
        proxy_send_timeout: 600s
        proxy_read_timeout: 600s
        names_hash_bucket_size: 128
    upstreams: {  }
    vhosts:
        somesite_vhost:
            server_name: mysite-test.somedomain
            www_root: /var/www/web
            listen_port: '80'
            index_files:
                - index.html
                - index.htm
                - index.php
            client_max_body_size: 1m
            ssl: '1'
            ssl_cert: ''
            ssl_key: ''
            ssl_port: '443'
            ssl_protocols: ''
            ssl_ciphers: ''
            rewrite_to_https: '1'
            spdy: '1'
            locations:
                nxvl_kelpcmbvphmp:
                    location: /
                    try_files:
                        - $uri
                        - '@rewriteapp'
                nxvl_u13asjfbasf3:
                    location: '@rewriteapp'
                    rewrite_rules:
                        - '^(.*)$ /app.php/$1 last'
                nxvl_vpo76zm42276:
                    location: '~ \.php(/|$)'
                    fastcgi: '127.0.0.1:9000'
                    fastcgi_index: index.php
                    fastcgi_split_path: '^(.+\.php)(/.*)$'
                    fast_cgi_params_extra:
                        - 'SCRIPT_FILENAME $document_root$fastcgi_script_name'
                        - 'HTTPS $https'
                    set:
                        - '$path_info $fastcgi_path_info'
                    proxy: ''
                    proxy_redirect: ''
                    location_cfg_append:
                        fastcgi_read_timeout: 3600
    proxies: {  }
apache:
    install: '0'
    settings:
        user: www-data
        group: www-data
        default_vhost: true
        manage_user: false
        manage_group: false
        sendfile: 0
    modules:
        - proxy_fcgi
        - rewrite
    vhosts:
        av_h7xfmvwa8odb:
            servername: awesome.dev
            serveraliases:
                - www.awesome.dev
            docroot: /var/www/awesome
            port: '80'
            setenv:
                - 'APP_ENV dev'
            setenvif:
                - 'Authorization "(.*)" HTTP_AUTHORIZATION=$1'
            custom_fragment: ''
            ssl: '0'
            ssl_cert: ''
            ssl_key: ''
            ssl_chain: ''
            ssl_certs_dir: ''
            ssl_protocol: ''
            ssl_cipher: ''
            directories:
                avd_8gbomqo33iv2:
                    path: /var/www/awesome
                    options:
                        - Indexes
                        - FollowSymlinks
                        - MultiViews
                    allow_override:
                        - All
                    require:
                        - 'all granted'
                    custom_fragment: ''
                    files_match:
                        avdfm_tbzdfnqir3lg:
                            path: \.php$
                            sethandler: 'proxy:fcgi://127.0.0.1:9000'
                            custom_fragment: ''
                            provider: filesmatch
                    provider: directory
php:
    install: '1'
    settings:
        version: '54'
    modules:
        php:
            - cli
            - intl
            - mcrypt
            - imagick
            - mysql
            - sqlite
            - gd
        pear: {  }
        pecl: {  }
    ini:
        display_errors: 'On'
        error_reporting: '-1'
        error_log: /var/log/php_errors.log
        session.save_path: /var/lib/php/session
        date.timezone: UTC
        realpath_cache_size: '4096k'
        realpath_cache_ttl: 7200
    fpm_ini:
        error_log: /var/log/php-fpm.log
    fpm_pools:
        phpfp_sij7fyb3g1v9:
            ini:
                prefix: www
                listen: '127.0.0.1:9000'
                security.limit_extensions: .php
                user: www-user
                group: www-data
    composer: '1'
    composer_home: ''
xdebug:
    install: '1'
    settings:
        xdebug.default_enable: '1'
        xdebug.remote_autostart: '1'
        xdebug.remote_connect_back: '1'
        xdebug.remote_enable: '1'
        xdebug.remote_handler: dbgp
        xdebug.remote_port: '9000'
        xdebug.remote_host: '192.168.58.1'
        xdebug.remote_log: '/var/log/xdebug.log'
blackfire:
    install: '0'
    settings:
        server_id: ''
        server_token: ''
        agent:
            http_proxy: ''
            https_proxy: ''
            log_file: stderr
            log_level: '1'
        php:
            agent_timeout: '0.25'
            log_file: ''
            log_level: '1'
xhprof:
    install: '0'
wpcli:
    install: '0'
    version: v0.19.0
drush:
    install: '0'
    version: 6.3.0
ruby:
    install: '1'
    versions:
        rv_pfx86pz44cd9:
            default: '1'
            bundler: '1'
            version: 1.9.3
python:
    install: '1'
    packages: {  }
    versions: {  }
nodejs:
    install: '1'
    npm_packages:
        - less
        - bower
hhvm:
    install: '0'
    nightly: 0
    composer: '1'
    composer_home: ''
    settings: {  }
    server_ini:
        hhvm.server.host: 127.0.0.1
        hhvm.server.port: '9000'
        hhvm.log.use_log_file: '1'
        hhvm.log.file: /var/log/hhvm/error.log
    php_ini:
        display_errors: 'On'
        error_reporting: '-1'
        date.timezone: UTC
mariadb:
    install: '0'
    settings:
        version: '5.5'
        root_password: '123'
        override_options: {  }
    adminer: 0
    users:
        mariadbnu_eifga4pgb1vr:
            name: dbuser
            password: '123'
    databases:
        mariadbnd_xfcy7bysw2f0:
            name: dbname
            sql: ''
    grants:
        mariadbng_d3xwiqv92gmb:
            user: dbuser
            table: '*.*'
            privileges:
                - ALL
mysql:
    install: '1'
    settings:
        version: '5.5'
        root_password: Pr0gramm1ng!$
        override_options: {  }
    adminer: 0
    users:
        mysqlnu_mauwcrgxukjt:
            name: mysite
            password: mysite
    databases:
        mysqlnd_bznifx8gl44h:
            name: mysite
#            sql: '/var/www/db.sql'
    grants:
        mysqlng_up0jz1cm3bq0:
            user: mysite
            table: '*.*'
            privileges:
                - ALL
postgresql:
    install: '0'
    settings:
        global:
            encoding: UTF8
            version: '9.4'
        server:
            postgres_password: '123'
    databases: {  }
    users: {  }
    grants: {  }
    adminer: 0
mongodb:
    install: '0'
    settings:
        auth: 1
        bind_ip: 127.0.0.1
        port: '27017'
    globals:
        version: 2.6.0
    databases: {  }
redis:
    install: '0'
    settings:
        conf_port: '6379'
sqlite:
    install: '0'
    adminer: 0
    databases: {  }
mailhog:
    install: '0'
    settings:
        smtp_ip: 0.0.0.0
        smtp_port: 1025
        http_ip: 0.0.0.0
        http_port: '8025'
        path: /usr/local/bin/mailhog
beanstalkd:
    install: '0'
    settings:
        listenaddress: 0.0.0.0
        listenport: '11300'
        maxjobsize: '65535'
        maxconnections: '1024'
        binlogdir: /var/lib/beanstalkd/binlog
        binlogfsync: null
        binlogsize: '10485760'
    beanstalk_console: 0
rabbitmq:
    install: '0'
    settings:
        port: '5672'
    users: {  }
    vhosts: {  }
    plugins: {  }
elastic_search:
    install: '0'
    settings:
        version: 2.1.0
        java_install: true
solr:
    install: '0'
    settings:
        version: 4.10.2
        port: '8984'
```

**Comment by jonasmello** [comment]
I have the same problem

**Comment by josezenem** [comment]
Same issue here.

**Comment by Braunson** [comment]
Try disabling any firewall while `vagrant up` is running.. I had the same issue, is related to my firewall blocking outbound connections...

**Related thread with info on why it all of a sudden doesn't work:** https://github.com/puphpet/puphpet/issues/1850

**Comment by lucasgrecco** [comment]
Same issue here! Someone have any news about how to fix that?

**Comment by jsbba** [comment]
Same issue here

**Comment by mike-source** [comment]
Same issue, no firewall or proxy

**Comment by jtreminio** [comment]
Folks, this isn't a puphpet issue. Check your DNS , router configuration, anything. Nothing I can do on my end.

**Comment by mike-source** [comment]
Sorry... my bad, it was something I did, but still a little baffling.

I was toying around trying to setup nginx as a reverse proxy, which involved putting a file in `/files/exec-always/50-vhost.sh`

(see: https://github.com/puphpet/puphpet/pull/1587)

As soon as I removed that file, it stopped having the ruby gem error.

Why that happens I have no idea, but thought I'd let people know in case it potentially helps.
